### PR TITLE
solution was added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

**Review by vladkuzya — APPROVED** [review]
Nice work!

File: pom.xml [hunk 1]
```diff
@@ -34,6 +44,11 @@
             <artifactId>mysql-connector-java</artifactId>
             <version>8.0.21</version>
         </dependency>
+        <dependency>
+            <groupId>commons-logging</groupId>
+            <artifactId>commons-logging</artifactId>
+            <version>1.1.1</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: why so many dependencies? log4j would be enough

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -16,9 +16,11 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
+import java.util.logging.Logger;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: wrong class imported in all the projects

also, you are missing `log4j.properties` configuration file

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -16,9 +16,11 @@
...
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.dev.cinema");
+    private static Logger logger = Logger.getLogger(Main.class.getName());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: you should be possible to pass class, not a string, after you import the right Logger class

File: src/main/java/com/dev/cinema/Main.java [hunk 4]
```diff
@@ -76,7 +78,7 @@ public static void main(String[] args) throws AuthenticationException {
 
         OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
         orderService.completeOrder(shoppingCartService.getByUser(iryna).getTickets(), iryna);
-        System.out.println(shoppingCartService.getByUser(iryna));
-        System.out.println(orderService.getOrderHistory(iryna));
+        logger.info("Get order by user id: " + shoppingCartService.getByUser(iryna));
+        logger.info("Get order by user id: " + orderService.getOrderHistory(iryna));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 5]
```diff
@@ -6,12 +6,15 @@
...
 import com.dev.cinema.util.HibernateUtil;
 import java.util.List;
+import java.util.logging.Logger;
 import javax.persistence.criteria.CriteriaQuery;
 import org.hibernate.Session;
 import org.hibernate.Transaction;
 
 @Dao
 public class CinemaHallDaoImpl implements CinemaHallDao {
+    private static Logger logger = Logger.getLogger(CinemaHallDao.class.getName());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
```
same everywhere

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -24,6 +24,16 @@
             <artifactId>hibernate-core</artifactId>
             <version>5.4.21.Final</version>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.30</version>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: do you use this dependency?

File: src/main/java/com/dev/cinema/Main.java [hunk 7]
```diff
@@ -16,11 +16,13 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.LocalTime;
+import org.apache.log4j.Logger;
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.dev.cinema");
+    private static Logger logger = Logger.getLogger(Main.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]:
```suggestion
    private static final Injector injector = Injector.getInstance("com.dev.cinema");
    private static final Logger logger = Logger.getLogger(Main.class);
```

File: src/main/java/com/dev/cinema/Main.java [hunk 8]
```diff
@@ -73,10 +83,9 @@ public static void main(String[] args) throws AuthenticationException {
         shoppingCartService.registerNewShoppingCart(iryna);
         shoppingCartService.addSession(movieSession, iryna);
         shoppingCartService.addSession(movieSessionSecond, iryna);
-
         OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
         orderService.completeOrder(shoppingCartService.getByUser(iryna).getTickets(), iryna);
-        System.out.println(shoppingCartService.getByUser(iryna));
-        System.out.println(orderService.getOrderHistory(iryna));
+        logger.info("Get order by user id: " + shoppingCartService.getByUser(iryna));
+        logger.info("Get order by user id: " + orderService.getOrderHistory(iryna));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: reconsider these messages, I think they are confusing

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 9]
```diff
@@ -7,11 +7,14 @@
 import com.dev.cinema.util.HibernateUtil;
 import java.util.List;
 import javax.persistence.criteria.CriteriaQuery;
+import org.apache.log4j.Logger;
 import org.hibernate.Session;
 import org.hibernate.Transaction;
 
 @Dao
 public class CinemaHallDaoImpl implements CinemaHallDao {
+    private static final Logger logger = Logger.getLogger(CinemaHallDao.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
    private static final Logger logger = Logger.getLogger(CinemaHallDaoImpl.class);
```
fix this in all the classes

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**